node/config/types.go

```diff
@@ -601,6 +601,16 @@ type Splitstore struct {
 	// A value of 0 disables, while a value 1 will do full GC in every compaction.
 	// Default is 20 (about once a week).
 	HotStoreFullGCFrequency uint64
+	// HotStoreMaxSpaceTarget sets a target max disk size for the hotstore. Splitstore GC
+	// will run moving GC if disk utilization gets within a threshold (150 GB) of the target.
+	// Splitstore GC will NOT run moving GC if the total size of the move would get
+	// within 50 GB of the target, and instead will run a more aggressive online GC.
```

Comment:
Isn't the move basically always going to be larger than 100GiB tho? In which case it would never run because we'd always already be withing 50G of the target

Reply:
Move is always going to be larger than 100GiB but it should still run, and I observe it running on my mainnet node with the current code.  The target is for the entire hotstore size.  So if the Newly Moved + Current sizes are too close (within 50GB) to the target then move won't run.  But if Newly Moved + Current are within 150 GB of the target but not 50 GB to the target than move runs.

Example 1 you have a hotstore of size 300G and a target of 800G, compaction computes a totally GCed size of 250G.  Then 300 + 250 G = 550 < 800 - 150 = 650, so move won't run.

Example 2 same sizes but target is 650, then 650 - 150 = 500 < 550 so move will try to run.  Move is able to run within safety buffer because 650 - 50 = 600 > 550.

Example 3 target is 599.  Move now can't run because total moving size 550 > 559 - 50.  So we will run aggressive online GC in hopes of getting hotstore size down back into a safe region.

Reply:
I see..

This should probably work, I guess we'll see in real world deployments.

Actually it would be really good to make it possible to configure targetThreshold / targetBuffer, ideally through config, but envvars would also work for now